### PR TITLE
API Policy integration tests

### PR DESCRIPTION
**Pull request by renuka-fernando**
### Purpose
- Add new echo resource in mock backend service which contains path, HTTP method
- Add tests for API policies
- Add tests for call interceptor service which uses the existing interceptor test cases.  Undeploy the APIs that used for interceptor service test cases, and deploy another APIs with same resources and same interceptor configs using API policies
- OPA policy integration tests
- Add support to mount volumes for integration test CC setup

API Policies
- Add header
- Remove header
- Add query param
- Call interceptor service
- Change HTTP method
- Path rewrite
- OPA policy

### Issues

Fixes https://github.com/wso2/product-microgateway/issues/2742
Fixes https://github.com/wso2/product-microgateway/issues/2741

### Automation tests
 - Unit tests added: No
 - Integration tests added: Yes

### Tested environments

Integration Tests executed in local

---
#### Maintainers: Check before merge
- [x] Assigned 'Type' label
- [x] Assigned the project
- [x] Validated respective github issues
- [x] Assigned milestone to the github issue(s)

### CODE REVIEW

File: integration/test-integration/src/test/resources/testng-cc-standalone.xml [hunk 1]
```diff
@@ -60,22 +60,17 @@
...
             <class name="org.wso2.choreo.connect.tests.testcases.standalone.interceptor.InterceptorServiceRequestBodyTestCase"/>
-        </classes>
-    </test>
-    <test name="cc-with-jwt-config" preserve-order="true" parallel="false">
-        <classes>
-            <class name="org.wso2.choreo.connect.tests.setup.standalone.CcWithJwtConfig"/>
-            <class name="org.wso2.choreo.connect.tests.testcases.standalone.security.CorsTestCase"/>
-            <class name="org.wso2.choreo.connect.tests.testcases.standalone.jwtGenerator.JwtGeneratorTestCase"/>
-            <class name="org.wso2.choreo.connect.tests.testcases.standalone.jwtGenerator.JwtTransformerTestCase"/>
-            <class name="org.wso2.choreo.connect.tests.testcases.standalone.timeouts.TimeoutTestCase"/>
```
Comment (suksw) [inline integration/test-integration/src/test/resources/testng-cc-standalone.xml:78]: Shown in red?

File: integration/mock-backend-server/src/main/java/org/wso2/choreo/connect/mockbackend/Utils.java [hunk 2]
```diff
@@ -45,6 +51,30 @@ public static void echo(HttpExchange exchange) throws IOException {
...
+                .collect(Collectors.toMap(q -> q[0], q -> q[1]));
+        echoResponse.setQuery(queryMap);
+
+        Gson gson = new Gson();
+        byte[] response = gson.toJson(echoResponse).getBytes();
+        int respCode = response.length == 0 ? HttpURLConnection.HTTP_NO_CONTENT : HttpURLConnection.HTTP_OK;
+        exchange.getResponseHeaders().set(Constants.CONTENT_TYPE, Constants.CONTENT_TYPE_APPLICATION_JSON);
+        exchange.sendResponseHeaders(respCode, response.length);
+        exchange.getResponseBody().write(response);
+        exchange.close();
```
Comment (suksw) [inline integration/mock-backend-server/src/main/java/org/wso2/choreo/connect/mockbackend/Utils.java:75]:
```suggestion
        respondWithBodyAndClose(respCode, response, exchange)
```

File: integration/test-integration/src/test/java/org/wso2/choreo/connect/tests/context/ChoreoConnectImpl.java [hunk 3]
```diff
@@ -211,4 +211,18 @@ public static void addInterceptorCertToRouterTruststore() throws IOException {
                     StandardOpenOption.APPEND);
         }
     }
+
+    public static void addCertsToEnforcerTruststore(String certsDir) throws CCTestException {
+        String certsDirPath = ChoreoConnectImpl.class.getClassLoader()
+                .getResource("certs/" + certsDir).getPath();
+        Utils.copyDirectory(certsDirPath, Utils.getTargetDirPath() + File.separator + TestConstant.CC_TEMP_PATH
```
Comment (suksw) [inline integration/test-integration/src/test/java/org/wso2/choreo/connect/tests/context/ChoreoConnectImpl.java:218]:
```suggestion
        Utils.copyDirectory(certsDirPath, Utils.getTargetDirPath() + TestConstant.CC_TEMP_PATH
```

**Review by suksw — APPROVED** [review]

File: integration/mock-backend-server/src/main/java/org/wso2/choreo/connect/mockbackend/Utils.java [hunk 4]
```diff
@@ -45,6 +51,30 @@ public static void echo(HttpExchange exchange) throws IOException {
         exchange.close();
     }
 
+    // echo request body, request headers in echo response payload
+    public static void echoInsidePayload(HttpExchange exchange) throws IOException {
```
[the remark this answers is not in the document]
Comment (renuka-fernando) [reply integration/mock-backend-server/src/main/java/org/wso2/choreo/connect/mockbackend/Utils.java]: Renamed with https://github.com/wso2/product-microgateway/pull/2743/commits/af1d7b8bbf4585d0f8c158e4d5af71fab7a28726

File: integration/mock-backend-server/src/main/java/org/wso2/choreo/connect/mockbackend/Utils.java (hunk 2 above)
Comment (renuka-fernando) [reply integration/mock-backend-server/src/main/java/org/wso2/choreo/connect/mockbackend/Utils.java]: Fixed with https://github.com/wso2/product-microgateway/pull/2743/commits/af1d7b8bbf4585d0f8c158e4d5af71fab7a28726

File: integration/test-integration/src/test/java/org/wso2/choreo/connect/tests/context/ChoreoConnectImpl.java (hunk 3 above)
Comment (renuka-fernando) [reply integration/test-integration/src/test/java/org/wso2/choreo/connect/tests/context/ChoreoConnectImpl.java]: Fixed with https://github.com/wso2/product-microgateway/pull/2743/commits/af1d7b8bbf4585d0f8c158e4d5af71fab7a28726

**Comment by renuka-fernando** [comment]
Added new commit https://github.com/wso2/product-microgateway/pull/2743/commits/af1d7b8bbf4585d0f8c158e4d5af71fab7a28726 to fix https://github.com/wso2/product-microgateway/issues/2741

**Review by pubudu538 — APPROVED** [review]

**Pull request merged**